arangod/Graph/GraphCache.cpp

```diff
@@ -0,0 +1,188 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#include "GraphCache.h"
+
+#include "Graph.h"
+
+#include <velocypack/Buffer.h>
+#include <velocypack/Collection.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/velocypack-aliases.h>
+#include <array>
+#include <boost/variant.hpp>
+#include <utility>
+
+#include "Aql/AstNode.h"
+#include "Aql/Graphs.h"
+#include "Aql/Query.h"
+#include "Basics/ReadLocker.h"
+#include "Basics/StaticStrings.h"
+#include "Basics/VelocyPackHelper.h"
+#include "Basics/WriteLocker.h"
+#include "RestServer/QueryRegistryFeature.h"
+#include "Transaction/Methods.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/OperationOptions.h"
+#include "Utils/SingleCollectionTransaction.h"
+#include "VocBase/Graphs.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/Methods/Collections.h"
+
+using namespace arangodb;
+using namespace arangodb::graph;
+
+namespace getGraphFromCacheResult {
+struct Success {
+  std::shared_ptr<Graph const> graph;
+
+  explicit Success(std::shared_ptr<Graph const> graph_)
+      : graph(std::move(graph_)){};
+  Success& operator=(Success const& other) = default;
+  Success() = delete;
+};
+struct Outdated {};
+struct NotFound {};
+struct Exception {};
+}
+
+using GetGraphFromCacheResult = boost::variant<
+    getGraphFromCacheResult::Success, getGraphFromCacheResult::Outdated,
+    getGraphFromCacheResult::NotFound, getGraphFromCacheResult::Exception>;
+
+GetGraphFromCacheResult _getGraphFromCache(GraphCache::CacheType const& _cache,
```

Comment:
According to our coding conventions, method names do not start with an underscore.